### PR TITLE
You can't talk on the radio while handcuffed or while stunned in some form.

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request

Helps encourage stealthy antagonist gameplay by preventing ;HELP SCI MAINT while you're being murdered due to the lack of -GLORF, while also evening the gap between fast and slow typers.

Also fixes ;SHITSEC.

## Why It's Good For The Game
Helps encourage stealthy antagonist gameplay by preventing ;HELP SCI MAINT while you're being murdered due to the lack of -GLORF, while also evening the gap between fast and slow typers.

Also fixes ;SHITSEC.

## Changelog
:cl:
balance: You can't talk on the radio while handcuffed or while stunned in some form.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
cult shackles explicitly mute you so you probably want to remove that

**Comment by willox** [comment]
Does this stop people from using the radio while in a clown car?

**Comment by Iamgoofball** [comment]
@willox That depends on what exactly clown cars do to their ~~victims~~passengers.

**Comment by SuperNovaa41** [comment]
This makes no sense, nor do you actually justify it

**Comment by CameronWoof** [comment]
We currently have radio jammers, which cost TC to purchase and are contraband to own, which serves the purpose of enabling stealth antagonists to kidnap people without them being able to call for help. I don't agree with the ability of that item being rolled in to handcuffs automatically for no cost.

**Review by Rohesie — CHANGES_REQUESTED** [review]
I'm not married to this idea, but if implemented properly and not blocked by other maintainers I'm willing to give it a try.
The current implementation is not proper, though.

Don't use `TRAIT_HANDS_BLOCKED`, which is tied to a ton of behavior you probably don't want to carry along (such as any sealed vehicle including the clown car blocking speech, aggressive grabs and xeno nests as well), and it might get linked to even more in the future.
Instead, make a new trait, something like `TRAIT_EARS_USAGE_BLOCKED`, and add/remove as necessary. You can find all the 24 instances of adding the other trait with the `ADD_TRAIT\(.*TRAIT_HANDS_BLOCKED` regex, and you'll notice that most don't need to block ears usage.

Also, instead of handling things in `/mob/living/carbon/radio()` and `talk_into()` both you should handle it in `/mob/living/carbon/human/radio()`

EDIT: I'm having second thoughts about this. I'm not sure if this is really compatible with the kind of gameplay loop we have in the LRP servers.

**Comment by Iamgoofball** [comment]
shit i hit the wrong button

**Comment by Iamgoofball** [comment]
>I'm not sure if this is really compatible with the kind of gameplay loop we have in the LRP servers.

We do not care about players at LRP, they should be playing at MRP standards because oranges has declared us an MRP codebase.

**Comment by Donpedrito** [comment]
Antagonists already have a number of tools at their disposal to prevent being called out over the radio. I’m not a fan of taking cult mute, muzzles, radio jammer, mute toxin, mute sting, disabling telecomms with TC, disabling telecomms with a bomb, abductor batons, and other tools and rendering them mostly useless because their functionality has been given to the funny stunbaton.

**Comment by moocowswag** [comment]
Stun combat doesn`t need another massive direct buff, one of the main fundamental counters to it is receiving outside assistance since you will FULLY recover on your own as long as you`re protected in your incapacitated state.

This just enables antags to craft/steal stun batons and go around baton cuffing people and dragging them into maint to be killed without needing tcomms to be down or countered at all.

**Comment by D4C-420** [comment]
There's already a way all people have to prevent talking over comms. its called stealing their headset which anyone can do without much effort or prep.

**Comment by Thunder12345** [comment]
Or dropping a couple of TCs on a radio jammer for an easy, passive, AoE comms blackout

**Comment by Ryll-Ryll** [comment]
this doesn't seem like a good idea for either LRP and MRP

**Comment by LemonInTheDark** [comment]
I'm sorry roh, I don't want this. Violently.

Most of the fun of interactions in this game are tied to interacting with the sandbox. 
What you're doing here invalidates much of that, and makes many traitor items and general tactics pointless.

If the issue you're trying to address is people not using those tools, the solution is not to come in with a bulldozer make a expansive system totally irrelevant.
I really hate this kind of thing. Regardless of if people are radioing too much or not (not that I would know), this shit absolutely makes the game worse. I realize making a more in depth change is more tricky, but I will not live with this. Please don't do it in future, you're better then it.

**Comment by LemonInTheDark** [comment]
Leaving this open for now, will close at 24 hours unless a maint comes for me

**Comment by moocowswag** [comment]
> There's already a way all people have to prevent talking over comms. its called stealing their headset which anyone can do without much effort or prep.

If you can`t yell "help [insert location here]" In the time it takes for someone to stun you and watch a bar fill up as they take off your radio then thats kind of on you. 

> Or dropping a couple of TC

Exactly, youre paying to be able to do that, this pr makes it so that if you have a stun baton or flash or any other stun weapon you automatically have the additional advantage of instantly preventing your victim from talking as soon as you apply that weapon without needing to disable tcomms or purchasing something...

**Comment by ANAME593** [comment]
think about it tho if somebody cuffs you what stops you from talking into a headset same with being stamcritted by batons or disabler beams

**Comment by cacogen** [comment]
What the fuck did the lizardpeople in charge of the code nowadays do to glorf anyway? I haven't seen it in a long time so I'm guessing they removed it. That was one of those additions of the future you never expected to see in the game. Like how does it know you're typing? It's a client side interaction with a text box and yet it does and it'll candlejack tha

**Comment by LemonInTheDark** [comment]
We can't do it without the command bar, which is not what most people use

**Comment by cacogen** [comment]
We can rebuild him https://github.com/tgstation/tgstation/pull/52341#issuecomment-660748051

**Comment by Thunder12345** [comment]
> We can rebuild him [#52341 (comment)](https://github.com/tgstation/tgstation/pull/52341#issuecomment-660748051)

Better get coding

**Comment by AnCopper** [comment]
ITT: People do not know how manual radios work.

**Comment by cacogen** [comment]
Yeah it's not voice activated you have to push your ear in like a secret service agent and the reason you can do this with both hands full is because you just use your knuckle stupid. You raise the entire potted plant to the side of your head and push on the headset with the back of your hand dummy.

**Comment by Rohesie** [comment]
24 hours is up, this is being closed due to not being wanted by the maintainers.
Please get a maintainer to support it before re-opening it.

**Comment by Addust** [comment]
> > I'm not sure if this is really compatible with the kind of gameplay loop we have in the LRP servers.
> 
> We do not care about players at LRP, they should be playing at MRP standards because oranges has declared us an MRP codebase. I'm not entirely convinced that we should be catering to players who already refuse to use communication outside of validhunting.

terry will hate you

**Comment by Rohesie** [comment]
Shaps expressed dislike and Potato blocked it.
Even if other maintainers agree, they'll have to convince Potato.
And, honestly, I'm not sure it's a good idea either.

**Comment by Iamgoofball** [comment]
Maybe actually give it a shot for a few weeks and collect data?

**Comment by Rohesie** [comment]
Try convincing Potato and Shaps.

**Comment by Mothblocks** [comment]
> We do not care about players at LRP, they should be playing at MRP standards because oranges has declared us an MRP codebase.

this is not what the codebase is MRP means. there is a reason it is always followed up with the sentence "manuel is HRP"

**Pull request closed without merging**
